rocketpy/Flight.py

```diff
@@ -2645,306 +2083,177 @@ def angleOfAttack(self, interpolation="spline", extrapolation="natural"):
 
         # Calculate angle of attack and convert to degrees
         angleOfAttack = np.rad2deg(np.arccos(dotProductNormalized))
-        angleOfAttack = np.column_stack([self.x[:, 0], angleOfAttack])
-
-        # Convert to Function
-        angleOfAttack = Function(
-            angleOfAttack,
-            "Time (s)",
-            "Angle Of Attack (°)",
-            interpolation,
-            extrapolation,
-        )
+        angleOfAttack = np.column_stack([self.time, angleOfAttack])
 
         return angleOfAttack
 
     # Stability and Control variables
+    def _frequency_analysis(self, func, samplingFrequency=1000):
```

Comment:
Not really sure... I guess I just thought the higher the better.

Reply:
@giovaniceotto , I ask again, don't you thnik this method should receive double underscore? This way you would create a non public method. Otherwise it just feels weird the method 's name

Reply:
Sorry, hadn't noticed the question. I did not want the method to be private because it can be used with any Function. Using a single underscore means it is meant to be used internally (therefore if we change it later, it is not a breaking change), but can also be used outside of the class by other functions and other classes (think analysis class).

I would prefer one of the following:
1. Keep it this way;
2. Make it public, by removing the leading underscore;
3. Transfer its functionality into the Function class (Function.toFrequencyDomain perhaps).

The third option is out of scope in relation to this PR. If you chose one of the other two, we could make the change and move forward.

Reply:
I'm sorry to bother you again, but I don't fell too happy with this method yet.
Seems to me a kind impossible to use it by another class, using the curret implementation, as you can only use it if you create a Flight object

**Some considerations here:**
1. Consider the recently #283 comments, the appropiate place for functions that can be used by different classes is tools.py. 
2. However, at liquid motors project, they are using supplement.py instead.
3. Yeah, confusion and confusion!

**My suggestions:** 
**Option A**: remove the depency of self.outOfRailTime attribute, wich is the only class attribute accessed by this method. We can set this as an argument and therefore the method is ready to become a `@staticmethod`, without depending on self as an argument.

**Option B**: Just move it to tools.py and invoke here at function.

Reply:
Unfortunately, I was really confusing when explaining my motives. An analysis class can make use of this method to get the frequency analysis of any Flight class attribute. I did not mean that it could use to perform frequency analysis of other signals.

I will follow your suggestion and remove the dependency of the `Flight.outOfRailTime` attribute and create a new commit.

**Quick note**:

> However, at liquid motors project, they are using supplement.py instead.

As of #299, this is not the case anymore.